src/MainWindow.cpp

```diff
@@ -493,12 +493,23 @@ MainWindow::MessageReceived(BMessage* msg)
 			}
 
 			TransactionData data;
-			gDatabase.GetTransaction(acc->CurrentTransaction(), data);
+			gDatabase.GetTransaction(acc->CurrentTransaction(), acc->GetID(), data);
 
-			if (data.Type().TypeCode() == TRANS_NUMERIC) {
-				ShowAlert(B_TRANSLATE("Numbered transactions cannot be scheduled"),
-					B_TRANSLATE("You can schedule transfers, deposits, or ATM transactions."));
-				break;
+			if (data.Type().TypeCode() == TRANS_XFER) {
+				if (data.Amount().IsPositive()) {
+					ShowAlert(B_TRANSLATE("Scheduling error"),
+						B_TRANSLATE(
+							"Transfers can only be scheduled from the originating account. This "
+							"appears to be the destination account for the transfer."));
+					break;
+				}
+
+				int32 destAccount = gDatabase.GetTransferDestination(data.GetID(), acc->GetID());
+				if (destAccount < 0) {
+					ShowAlert(B_TRANSLATE("Scheduling error"),
+						B_TRANSLATE("Transfers generated by scheduling cannot be scheduled."));
```

Comment:
I think adding "themselves" would make it even clearer:

```suggestion
						B_TRANSLATE("Transfers generated by scheduling cannot be scheduled themselves."));
```

Reply:
Yes, that's better. Added.